### ISSUE TITLE
Business Units not checked after first contact

### ISSUE THREAD

**Issue by childrss** [munkireport/munkireport-php#854]
After a machine checks in with MunkiReport, the machine group key is no longer checked against what business unit it is affiliated with.

We discovered this using a VM testing to make sure a machine logged into various business units based on the installers we built -- if the machine

The solution within the interface is to delete the computer from MunkiReport, and when it contacts the report server it creates a new entry for the computer in the correct spot.

This is undesirable because you lose a lot of information (e.g. we were thinking of having our Authorized Apple Repair service have access to MR, and log any repairs.  Then when you see a machine at the Campus Surplus, you could look it up and see that it perhaps isn't a machine worth the trouble)!  However there could be information that needs to be sanitized prior to moving.

**Comment by bochoven** [comment]
Not sure what you are reporting here: are you able to move a machine to another machine group by changing the passphrase?

**Comment by childrss** [comment]
Nope, what you describe is close to the behavior we expected to see:  change the passphrase to a machine group code that is associated with a different business unit, and we expected the machine to show up in the new business unit the code is assigned to.

That didn't happen until we deleted the computer entry in munkireport.  So it appears that the business unit is set the first time the machine logs in and creates an entry, but after that the association isn't verified again.

**Comment by bochoven** [comment]
Something else must go wrong: machines have no notion of business units. They are only associated with a particular machine group. The munkireport admin handles which machine group belongs to which business unit.

**Comment by bochoven** [comment]
I can't replicate your issue. Moving a machine from one machine group to another works here:

```bash
mb-039430:~ root# defaults write /Library/Preferences/MunkiReport.plist Passphrase 94123DBA-8A8B-4B9A-93F1-7CACD3604226
mb-039430:~ root# /usr/local/munki/preflight.d/submit.preflight 
mb-039430:~ root# defaults write /Library/Preferences/MunkiReport.plist Passphrase 8F4C7753-884B-4B86-A4C4-E6D74747F7A1
mb-039430:~ root# /usr/local/munki/preflight.d/submit.preflight 
```
